src/ProtocolFeeController.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Ownable2Step, Ownable} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {PoolKey} from "./types/PoolKey.sol";
+import {PoolId} from "./types/PoolId.sol";
+import {Currency} from "./types/Currency.sol";
+import {LPFeeLibrary} from "./libraries/LPFeeLibrary.sol";
+import {IProtocolFeeController} from "./interfaces/IProtocolFeeController.sol";
+import {IProtocolFees} from "./interfaces/IProtocolFees.sol";
+import {ProtocolFeeLibrary} from "./libraries/ProtocolFeeLibrary.sol";
+
+/// @notice ProtocolFeeController for both Pool type
+contract ProtocolFeeController is IProtocolFeeController, Ownable2Step {
+    using ProtocolFeeLibrary for uint24;
+
+    /// @notice throw when the pool manager saved does not match the pool manager from the pool key
+    error InvalidPoolManager();
+
+    /// @notice throw when the protocol fee split ratio is invalid i.e. greater than 100%
+    error InvliadProtocolFeeSplitRatio();
+
+    /// @notice 100% in hundredths of a bip
+    uint256 public constant ONE_HUNDRED_PERCENT_RATIO = 1e6;
+
+    /// @notice The ratio of the protocol fee in the total fee, expressed in hundredths of a bip i.e. 1e4 is 1%
+    /// @dev The default value is 33% i.e. protocol fee should be 33% of the total fee
+    uint256 public protocolFeeSplitRatio = 33 * 1e4;
+
+    address public immutable poolManager;
+
+    /// @notice emit when the protocol fee is collected
+    event ProtocolFeeCollected(Currency indexed currency, uint256 amount);
+
+    constructor(address _clPoolManager) Ownable(msg.sender) {
```

Comment:
should be `_poolManager`? since we would deploy 1 protocolFeeController for each pool manager